### ISSUE TITLE
bundle adapters in `hattip`

### ISSUE THREAD

**Issue by samuelstroschein** [hattipjs/hattip#56]
## Problem 

1. Hattip only has a few hundred downloads on NPM making the decision to adopt hattip harder.
2. Adapters are separate packages. Developers have to download multiple `hattip` dependencies slightly lowering DX but, more importantly, spreading the downloads of Hattip across multiple packages. 

## Sugesstion

Bundle all adapters in `hattip` like `hattip/adapter/node`. 

- the npm downloads of all hattip packages are bundled, leading to more downloads and thereby easing adoption
- the deployment pipeline becomes easier because only one package needs to be deployed

We had a similar discussion for inlang https://github.com/inlang/inlang/pull/460#discussion_r1143781116

**Comment by brillout** [comment]
We've plans to significantly increase the number of HatTip users. I think the number of npm downloads won't be an issue then.

**Comment by cyco130** [comment]
> The CLI won't address the outlined problems :)

Yes, but the `hattip` package will. My wires got crossed between this and [the other conversation](https://github.com/hattipjs/hattip/issues/55) :)

**Comment by brillout** [comment]
Vite handles linked npm packages as `ssr.noExternal`, so AFAICT HMR should work just fine with your setup.
